edge-modules/TestAnalyzer/ReportingCacheWithStorage.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestAnalyzer
+{
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class ReportingCacheWithStorage
+    {
+        Storage storage;
+
+        public static ReportingCacheWithStorage Instance { get; } = new ReportingCacheWithStorage();
+
+        ReportingCacheWithStorage()
+        {
+        }
+
+        public async Task Init(string storagePath, bool optimizeForPerformance)
+        {
+            this.storage = new Storage();
+            await this.storage.Init(storagePath, new SystemEnvironment(), optimizeForPerformance);
+            await this.storage.ProcessAllMessages(message => ReportingCache.Instance.AddMessage(message));
```

Comment:
is it possible to run ProcessAllMessages, ProcessAllDirectMethods, and ProcessAllTwins in parallel and await When.All.